lib/source/defender/aws_iot_defender_mqtt.c

```diff
@@ -113,19 +115,31 @@ void AwsIotDefenderInternal_DeleteTopicsNames( void )
 
 /*-----------------------------------------------------------*/
 
-IotMqttError_t AwsIotDefenderInternal_MqttConnect( void )
+void AwsIotDefenderInternal_SetMqttConnection( IotMqttConnection_t mqttConnection )
 {
-    return IotMqtt_Connect( &_startInfo.mqttNetworkInfo,
-                            &_startInfo.mqttConnectionInfo,
-                            _defenderToMilliseconds( AWS_IOT_DEFENDER_MQTT_CONNECT_TIMEOUT_SECONDS ),
-                            &_mqttConnection );
+    _mqttConnection = mqttConnection;
 }
 
 /*-----------------------------------------------------------*/
 
 IotMqttError_t AwsIotDefenderInternal_MqttSubscribe( IotMqttCallbackInfo_t acceptCallback,
-                                                     IotMqttCallbackInfo_t rejectCallback )
+                                                     IotMqttCallbackInfo_t rejectCallback,
+                                                     IotMqttCallbackInfo_t * pMqttCallbackInfo )
 {
+    IOT_FUNCTION_ENTRY( IotMqttError_t, IOT_MQTT_SUCCESS );
+    IotMqttOperation_t subscribeOperation = IOT_MQTT_OPERATION_INITIALIZER;
+
+    /* If it is mocked mqtt connection return success */
+    if( _mqttConnection == DEFENDER_MQTT_CONNECTION_MOCK )
```

Comment:
See the already-implemented MQTT mocks... Library code should not have test-specific code in it!